pkg/azure/keyvault/store_test.go

```diff
@@ -0,0 +1,64 @@
+package keyvault
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"get.porter.sh/plugin/azure/pkg/azure/azureconfig"
+	"github.com/cnabio/cnab-go/secrets/host"
+	"github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/require"
+)
+
+func TestResolve_NonSecret(t *testing.T) {
+	logger := hclog.New(&hclog.LoggerOptions{
+		Name:   PluginInterface,
+		Output: os.Stderr,
+		Level:  hclog.Error,
+	})
+
+	azConfig := azureconfig.Config{}
+	store := NewStore(azConfig, logger)
+
+	t.Run("resolve non-secret source: value", func(t *testing.T) {
+		resolved, err := store.Resolve(host.SourceValue, "myvalue")
+		require.NoError(t, err)
+		require.Equal(t, "myvalue", resolved)
+	})
+
+	t.Run("resolve non-secret source: env", func(t *testing.T) {
+		os.Setenv("MY_ENV_VAR", "myvalue")
+		defer os.Unsetenv("MY_ENV_VAR")
+
+		resolved, err := store.Resolve(host.SourceEnv, "MY_ENV_VAR")
+		require.NoError(t, err)
+		require.Equal(t, "myvalue", resolved)
+	})
+
+	t.Run("resolve non-secret source: path", func(t *testing.T) {
+		file, err := ioutil.TempFile("", "myfile")
```

Comment:
The Azure Plugin has a porter context object, if we make `getPlugins` a receiver we can pass it through to them. 

```
	var opts RunOptions
        // ! Switch to passing the entire plugin to validate !
	err = opts.Validate(args, p)
	if err != nil {
		logger.Error(err.Error())
		return
	}

	plugins.Serve(opts.selectedInterface, opts.selectedPlugin)
}

// ! Change to a receiver !
func (p *Plugin) getPlugins() map[string]func() plugin.Plugin {
	return map[string]func() plugin.Plugin{
                // ! pass in the context as needed 
 		blob.PluginInterface:     func() plugin.Plugin { return blob.NewPlugin(p.Config) },
		keyvault.PluginInterface: func() plugin.Plugin { return keyvault.NewPlugin(p.Context, p.Config) },
	}
}
```

Reply:
Please ignore this, it's not feasible because as we discussed offline, cnab-go hits the os filesystem directly...